lighthouse-core/gather/driver/target-manager.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+  * @fileoverview This class tracks multiple targets (the page itself and its OOPIFs) and allows consumers to
+  * listen for protocol events before each target is resumed.
+  */
+
+const log = require('lighthouse-logger');
+
+/** @typedef {{target: LH.Crdp.Target.TargetInfo, session: LH.Gatherer.FRProtocolSession}} TargetWithSession */
+
+class TargetManager {
+  /** @param {LH.Gatherer.FRProtocolSession} session */
+  constructor(session) {
+    this._enabled = false;
+    this._session = session;
```

Comment:
it feels like if the intention is we have a root session that maybe we should start naming it something special since it's not like the other sessions, but no rush to even think about that right now.